sandboxes/staticanalysis/staticanalyze.go

```diff
@@ -0,0 +1,235 @@
+	if len(customParserPath) > 0 {
+		parserPath = customParserPath
+		parserType = "custom"
+	}
+
+	parserAbsPath, err := filepath.Abs(parserPath)
+	if err != nil {
+		return "", fmt.Errorf("could not resolve relative path %s for JS parser: %v", parserPath, err)
+	}
+	if _, err = os.Stat(parserAbsPath); err != nil {
+		return "", fmt.Errorf("could not locate %s JS parser at %s: %v", parserType, parserAbsPath, err)
+	}
+
+	return parserPath, nil
+}
+
+func checkAnalyses(names []string) ([]staticanalysis.Task, error) {
+	var tasks []staticanalysis.Task
+	for _, name := range names {
+		task, ok := staticanalysis.TaskFromString(name)
+		if !ok {
+			return nil, fmt.Errorf("unrecognised static analysis task: %s", name)
+		}
+		tasks = append(tasks, task)
+	}
+	return tasks, nil
+}
+
+func printAnalyses() {
+	println("Available analyses are:")
+	for _, task := range staticanalysis.AllTasks() {
+		println("\t", task)
+	}
+}
+
+func doObfuscationDetection(extractedPackageDir string) (*obfuscation.AnalysisResult, error) {
+	jsParserPath, err := getJsParserPath()
+	if err != nil {
+		return nil, err
+	}
+
+	result := &obfuscation.AnalysisResult{
+		FileRawData:    map[string]obfuscation.RawData{},
+		FileSignals:    map[string]obfuscation.Signals{},
+		PackageSignals: obfuscation.NoSignals(),
+		ExcludedFiles:  []string{},
+	}
+	err = filepath.WalkDir(extractedPackageDir, func(path string, f fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if f.Type().IsRegular() {
+			pathInArchive := strings.TrimPrefix(path, extractedPackageDir+string(os.PathSeparator))
+			fmt.Printf("Processing file %s\n", pathInArchive)
+			rawData, err := obfuscation.CollectData(jsParserPath, path, "", false)
+			if err != nil {
+				log.Error("Error parsing file", "filename", pathInArchive, "error", err)
+				result.ExcludedFiles = append(result.ExcludedFiles, pathInArchive)
+			} else if rawData == nil {
+				// syntax error - could not parse file
+				result.ExcludedFiles = append(result.ExcludedFiles, pathInArchive)
+			} else {
+				// rawData != nil, err == nil
+				result.FileRawData[f.Name()] = *rawData
+				signals := obfuscation.ComputeSignals(*rawData)
+				result.FileSignals[f.Name()] = obfuscation.RemoveNaNs(signals)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error while walking package files: %v", err)
+	}
+
+	return result, nil
+}
+
+func cleanupTmpDir(dir string) {
+	err := os.RemoveAll(dir)
+	if err != nil {
+		fmt.Printf("Failed to remove temp work dir: %v\n", err)
+	}
+}
+
+func makeWorkDirs() (tmpDir, archiveDir, extractDir string, err error) {
+	if tmpDir, err = os.MkdirTemp("", "package-analysis-staticanalyze"); err != nil {
+		return "", "", "", err
+	}
+
+	if archiveDir, err = os.MkdirTemp(tmpDir, "archive"); err != nil {
+		cleanupTmpDir(tmpDir)
+		return "", "", "", err
+	}
+
+	if extractDir, err = os.MkdirTemp(tmpDir, "extracted"); err != nil {
+		cleanupTmpDir(tmpDir)
```

Comment:
this needs to also cleanup `archiveDir` right?

Reply:
`archiveDir` is created inside `tmpDir` and the cleanup does an `os.RemoveAll()`, so it cleans everything up. Is that reasonable?

Reply:
Ah, got it. I think the interface will be a lot cleaner this way. It's not clear to callers that the first dir returned is the base dir that needs to be cleaned up. 

```golang
type WorkDirs struct {
  baseDir string
  ArchiveDir string
  ExtractDir string
}

func (wd *WorkDirs) Cleanup() {
  ...
}
```

Reply:
Sure I can do that